akka-docs/rst/java/code/docs/http/javadsl/server/directives/BasicDirectivesExamplesTest.java

```diff
@@ -0,0 +1,788 @@
+        if (rr instanceof akka.http.scaladsl.server.RouteResult.Complete) {
+          final HttpResponse res = ((akka.http.scaladsl.server.RouteResult.Complete) rr).getResponse();
+          return new akka.http.scaladsl.server.RouteResult.Complete(
+            (akka.http.scaladsl.model.HttpResponse)
+              res.addHeader(Server.create(ProductVersion.create("MyServer", "1.0"))));
+        } else {
+          return rr;
+        }
+      }), () -> complete("Hello world!"));
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/"))
+      .assertStatusCode(StatusCodes.OK)
+      .assertHeaderExists(Server.create(ProductVersion.create("MyServer", "1.0")));
+    //#mapRouteResultFuture
+  }
+
+//  @Test
+//  public void testMapRouteResult() {
+//    //#0mapRouteResult
+//    mapRouteResult(result ->)
+//    //#0mapRouteResult
+//  }
+
+  @Test
+  public void testMapResponseEntity() {
+    //#mapResponseEntity
+    final Route route = mapResponseEntity(entity -> {
+      if (entity instanceof HttpEntity.Strict) {
+        final HttpEntity.Strict strict = (HttpEntity.Strict) entity;
+        return HttpEntities.create(strict.getContentType(), ByteString.fromString("test").$plus$plus(strict.getData()));
+      } else {
+        throw new IllegalStateException("Unexpected entity type");
+      }
+    }, () -> complete("abc"));
+
+    testRoute(route).run(HttpRequest.GET("/"))
+      .assertEntity("testabc");
+    //#mapResponseEntity
+  }
+
+  @Test  public void testMapResponseHeaders() {
+    //#mapResponseHeaders
+    // adds all request headers to the response
+    final Route echoRequestHeaders = extract(
+      ctx -> ctx.getRequest().getHeaders(),
+      headers -> respondWithHeaders(headers, () -> complete("test"))
+    );
+
+    final Route route = mapResponseHeaders(headers -> {
+      headers.removeIf(header -> header.lowercaseName().equals("id"));
+      return headers;
+    }, () -> echoRequestHeaders);
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/").addHeaders(
+      Arrays.asList(RawHeader.create("id", "12345"),RawHeader.create("id2", "67890"))))
+      .assertHeaderKindNotExists("id")
+      .assertHeaderExists("id2", "67890");
+    //#mapResponseHeaders
+  }
+
+  @Ignore("Not implemented yet")
+  @Test
+  public void testMapInnerRoute() {
+    //#mapInnerRoute
+    // TODO: implement mapInnerRoute
+    //#mapInnerRoute
+  }
+
+  @Test
+  public void testMapRejections() {
+    //#mapRejections
+    // ignore any rejections and replace them by AuthorizationFailedRejection
+    final Route route = mapRejections(
+      rejections -> Collections.singletonList((Rejection) Rejections.authorizationFailed()),
+      () -> path("abc", () -> complete("abc"))
+    );
+
+    // tests:
+    runRouteUnSealed(route, HttpRequest.GET("/"))
+      .assertRejections(Rejections.authorizationFailed());
+    testRoute(route).run(HttpRequest.GET("/abc"))
+      .assertStatusCode(StatusCodes.OK);
+    //#mapRejections
+  }
+
```

Comment:
Make these three methods into lambdas assigned to variables where they are used, they are important for context when reading the examples in the docs.